tests/integration/pgo.rs

```diff
@@ -296,6 +296,35 @@ rustflags = ["-Ctarget-cpu=native"]
 "#,
     );
 
+    let output = project.cmd(&["build", "--", "-v"]).run()?;
+    println!("{}", output.stderr());
+    assert!(!output.stderr().contains("-Ctarget-cpu=native"));
```

Comment:
A very minor note, but this assumes the tests run on a system that hasn't set `-Ctarget-cpu=native` on a matching `target.rustflags`... which, of course, mine has, causing the test to fail.

Suggestion:

```diff
diff --git a/tests/integration/pgo.rs b/tests/integration/pgo.rs
index a60e19828f27..3a33ee5f8ca0 100644
--- a/tests/integration/pgo.rs
+++ b/tests/integration/pgo.rs
@@ -292,13 +292,13 @@ fn test_override_build_rustflags_from_config() -> anyhow::Result<()> {
         ".cargo/config.toml",
         r#"
 [build]
-rustflags = ["-Ctarget-cpu=native"]
+rustflags = ["-Ctarget-cpu=imaginary"]
 "#,
     );
 
     let output = project.cmd(&["build", "--", "-v"]).run()?;
     println!("{}", output.stderr());
-    assert!(!output.stderr().contains("-Ctarget-cpu=native"));
+    assert!(!output.stderr().contains("-Ctarget-cpu=imaginary"));
     assert!(output.stderr().contains("-Cprofile-generate"));
     output.assert_ok();
 ```

Reply:
This won't work, the test would fail because cargo does not know the `imaginary` target. It would be nice to disallow Cargo to load external config files, but I haven't found any way to do it.

Reply:
Hmm, it failed with `-Ctarget-cpu=foo` on my machine, by Rust saying that it doesn't know the `foo` target (as expected).

Aah, I see now, I replaced it everywhere, it should have only been done for this single test. Sorry.